src/Raven.Server/Web/System/SetupHandler.cs

```diff
@@ -0,0 +1,383 @@
+
+                    if (PlatformDetails.RunningOnPosix)
+                    {
+                        AdminCertificatesHandler.ValidateCaExistsInOsStores(node.Certificate, "Setup server certificate", ServerStore);
+                    }
+                }
+
+                // Prepare settings.json files, and write the local one to disk
+                var settingsPath = SetupManager.SettingsFileName;
+                var jsons = new Dictionary<string, string>();
+                SecuredSetupInfo.NodeInfo localNode = null;
+
+                foreach (var node in setupInfo.NodeSetupInfos)
+                {
+                    try
+                    {
+                        if (node.NodeTag == "A")
+                        {
+                            jsons.Add(node.NodeTag,
+                                SetupManager.WriteSettingsJsonFile(node.Certificate, node.PublicServerUrl, node.ServerUrl, settingsPath, SetupMode.Secured, modifyLocalServer: true));
+                            localNode = node;
+                        }
+                        else
+                            jsons.Add(node.NodeTag, SetupManager.WriteSettingsJsonFile(node.Certificate, node.PublicServerUrl, node.ServerUrl, settingsPath, SetupMode.Secured, modifyLocalServer: false));
+                    }
+                    catch (Exception e)
+                    {
+                        throw new InvalidOperationException($"Failed to update {settingsPath} with new configuration.", e);
+                    }
+                }
+
+                // Here we need to take the jsons, create files and zip them all together
+                // Need to make this an async operation, just like when sending a certificate for the studio... to know that we're done and sending a file
+                var zip = new byte[]{};
+                
+                try
+                {
+                    var ips = localNode?.Ips.Select(ip => new IPEndPoint(IPAddress.Parse(ip), localNode.Port)).ToArray();
+
+                    var certBytes = Convert.FromBase64String(localNode?.Certificate);
+                    var x509Certificate2 = new X509Certificate2(certBytes);
+                    ServerStore.SetupManager.AssertServerCanStartSecured(x509Certificate2, localNode?.ServerUrl, ips, settingsPath);
+
+                    // Load the certificate in the local server, so we can generate client certs later
+                    Server.ClusterCertificateHolder = SecretProtection.ValidateCertificateAndCreateCertificateHolder(localNode?.Certificate, "Setup", x509Certificate2, certBytes);
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException($"Failed to start server with the new configuration {settingsPath}.", e);
+
+                }
+
+                var contentDisposition = "attachment; filename=settings.zip";
```

Comment:
Filename here should be the first certificate public server url / server url of the first node with "Cluster Settings.zip"